src/Middleware/HeaderPropagation/src/HeaderPropagationEntry.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.HeaderPropagation
+{
+    /// <summary>
+    /// Define the configuration of a header for the <see cref="HeaderPropagationMiddleware"/>.
+    /// </summary>
+
+    public class HeaderPropagationEntry
+    {
+        /// <summary>
+        /// Gets or sets the name of the header to be used by the <see cref="HeaderPropagationMessageHandler"/> fot the outbound http requests.
+        /// </summary>
+        public string OutboundHeaderName { get; set; }
+
```

Comment:
I think somewhere in the docs it would be nice to have a comment like: 

```
If ValueFactory is null and DefaultValue is null then....
If ValueFactory is non-null then...
If ValueFactory is null and DefaultValues is non-null then...
```

Basically, somewhere we should document how users should think about the precedence of these options.